mcs/class/WindowsBase/ZipSharp/UnzipReadStream.cs

```diff
@@ -30,6 +30,7 @@ namespace zipsharp
 {
 	class UnzipReadStream : Stream
 	{
+		private static bool use32Bit = IntPtr.Size == 4 || Environment.OSVersion.Platform != PlatformID.Unix;
```

Comment:
Could you move all copies into a new class e.g. `static class NativeUnzipVersion` and add comment why it's done like that